backend/prompt_studio/prompt_studio_core/views.py

```diff
@@ -275,12 +283,14 @@ def single_pass_extraction(self, request: HttpRequest, pk: uuid) -> Response:
         custom_tool = self.get_object()
         tool_id: str = str(custom_tool.tool_id)
         document_id: str = request.data.get(ToolStudioPromptKeys.DOCUMENT_ID)
-
+        # Generate a run_id
+        run_id = str(uuid.uuid4())
```

Comment:
@gaya3-zipstack Let's introduce a utility function in prompt studio helper to generate a run ID and use this everywhere instead.

Reply:
Done